src/mscorlib/src/System/Threading/Tasks/Task.cs

```diff
@@ -6294,7 +6294,7 @@ private static Task<TResult[]> InternalWhenAll<TResult>(Task<TResult>[] tasks)
         {
             Contract.Requires(tasks != null, "Expected a non-null tasks array");
             return (tasks.Length == 0) ? // take shortcut if there are no tasks upon which to wait
-                new Task<TResult[]>(false, new TResult[0], TaskCreationOptions.None, default(CancellationToken)) :
+                new Task<TResult[]>(false, Array.Empty<TResult>(), TaskCreationOptions.None, default(CancellationToken)) :
```

Comment:
This is a very rare case.  To hit it, you'd need to be waiting for zero tasks, and for such cases if you think there's a chance you might have zero, the right optimization is to just not call WhenAll at all.  Given that, and since we don't know what TResult is and whether it's likely we're otherwise caching an empty array for it, I think we should revert this one, too.